crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -502,11 +536,11 @@ mod tests {
             Some(&functools_module),
             resolve_module(&db, functools_module_name).as_ref()
         );
-        assert_eq!(&src, functools_module.search_path().path());
+        assert_eq!(*src, functools_module.search_path());
```

Comment:
I recommend you to go once more througth all `*` usages and see if you can use `&` instead or don't need them all together.

Reply:
Thanks, I've done that. They were definitely necessary at some earlier point in this PR's history, but they no longer are. Annoying that Clippy didn't spot this :(